### ISSUE TITLE
a "No song playing" notification is shown when using Spotify - JQBX

### ISSUE THREAD

**Issue by ghost** [citruspi/Spotify-Notifications#112]
When playing Spotify using JQBX, a "No song playing" notification is displayed for every song change. The album art _is_ correct. When re-showing the notification (using the hotkey) the track and artist are there.

**Comment by mattbroussard** [comment]
I have noticed this too when playing through a Sonos. Perhaps it has something to do with Spotify Connect?

**Comment by oxydeon** [comment]
it's work again for me, no reason...

**Comment by mattbroussard** [comment]
It seems like this application talks to Spotify using Scripting Bridge, and Spotify is sometimes not returning anything for the "name" field:
```
$ osascript -e "tell application \"Spotify\" to get name of current track"

$ osascript -e "tell application \"Spotify\" to get id of current track"
spotify:track:1jILVSedO8KYfAZiXvRHcW
$ 
```

A possible workaround would be to separately look up the track using the ID/URL that seems to still be returned.

**Comment by shousper** [comment]
> I have noticed this too when playing through a Sonos. Perhaps it has something to do with Spotify Connect?

Seconded. Listening on my Sonos via Spotify right now and it's always showing "No song playing".

**Comment by jamescastells** [comment]
Spotify finally added support for Apple TV. However, when listening there, the notifications in my Mac show up as "No song playing".

**Comment by tonglil** [comment]
Same when listening through my Amazon Echo device, but it seems to only have started happening recently.

**Comment by rustysc** [comment]
Yeah like Matt said, it seems to be an issue with Spotify Connect
